components/script/dom/location.rs

```diff
@@ -223,8 +285,23 @@ impl LocationMethods for Location {
 
     // https://html.spec.whatwg.org/multipage/#dom-location-port
     fn SetPort(&self, value: USVString) -> ErrorResult {
-        self.check_same_origin_domain()?;
-        self.set_url_component(value, UrlHelper::SetPort);
+        // Step 1: If this Location object's relevant Document is null, then return.
+        if self.window.has_document() {
+            // Step 2: If this Location object's relevant Document's origin is not
+            // same origin-domain with the entry settings object's origin, then
+            // throw a "SecurityError" DOMException.
+            self.check_same_origin_domain()?;
+            // Step 3: Let copyURL be a copy of this Location object's url.
+            let copy_url = self.get_url();
+            // Step 4: If copyURL cannot have a username/password/port, then return.
+            // https://url.spec.whatwg.org/#cannot-have-a-username-password-port
```

Comment:
Similar re step 5, which could be done using `set_port`.

Reply:
There's one caveat though when setting port directly via `as_mut_url().set_port` - we'll have to manually verify and parse the input string to `Option<u16>`. However, we can use `url::quirks::set_port` to check and set the port here instead of `UrlHelper::SetPort`.

Reply:
Sounds good!